src/api/Endpoint.ts

```diff
@@ -348,12 +348,26 @@ export class Endpoint {
         controller: Controller, request: Request, response: Response, principal: Principal
     ) {
         let method: Function = controller[this.endpointInfo.methodName]
-        let parameters = this.buildEndpointParameters(request, response, principal)
-
         let controllerName = this.endpointInfo.controller ?
             this.endpointInfo.controller.name :
             "<dynamic>";
 
+        let parameters;
+
+        try {
+          parameters = this.buildEndpointParameters(request, response, principal)
+        } catch (ex) {
+          const errors = ex.reduce((list, error) => list.concat(Object.values(error.constraints)), []);
+          this.log(LogLevel.error, "Errors occurred extracting parameters for method '%s' in controller '%s': %j",
+                  this.endpointInfo.methodName,
+                  controllerName,
+                  errors)
+
+          response.status(400)
+          response.send({ errors })
```

Comment:
The `errors` object here contains validation errors, so IMO it's the integrator who needs to see the error (i.e. so they can fix their request to pass the validation), rather than the system owner, who can't do anything about it. I don't see a security concern in that, unless I suppose if `buildEndpointParameters` were to throw other errors. 

Perhaps I need to revisit the choice to throw the validator's `errors` object instead of just passing it back as part of a return object from `buildEndpointParameters`

Reply:
Sounds fair enough to me based on your explanation